kornia/geometry/camera/stereo.py

```diff
@@ -249,38 +269,28 @@ def reproject_disparity_to_3D(disparity_tensor: torch.Tensor, Q_matrix: torch.Te
     _check_Q_matrix(Q_matrix)
     _check_disparity_tensor(disparity_tensor)
 
-    batch_size, rows, cols = disparity_tensor.shape
+    batch_size, channels, rows, cols = disparity_tensor.shape
     dtype = disparity_tensor.dtype
     device = disparity_tensor.device
-    homogenous_observation_ndim = 4
-    euclidian_observation_ndim = homogenous_observation_ndim - 1
-
-    # Construct a mesh grid of uv values, i.e. the tensors will contain 1:rows and 1:cols.
-    u, v = torch.meshgrid(
-        torch.arange(rows, dtype=dtype, device=device), torch.arange(cols, dtype=dtype, device=device)
-    )
-    u, v = u.expand(batch_size, -1, -1), v.expand(batch_size, -1, -1)
 
-    # The z dimension in homogenous coordinates are just 1.
-    z = torch.ones((batch_size, rows, cols), dtype=dtype, device=device)
+    uv = create_meshgrid(rows, cols, normalized_coordinates=False, device=device, dtype=dtype)
+    v, u = uv[..., 0], uv[..., 1]
```

Comment:
```suggestion
    u, v = uv[..., 0], uv[..., 1]
```
`create_meshgrid` returns x,y

Reply:
Tip: unbind() is a faster way to unpack.